src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -80,8 +83,12 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     this.cyInitialization(this.getCy());
   }
 
-  componentDidUpdate() {
-    this.processGraphUpdate(this.getCy());
+  componentDidUpdate(prevProps: CytoscapeGraphProps, prevState: CytoscapeGraphState) {
+    const cy = this.getCy();
+    this.processGraphUpdate(cy);
+    if (this.props.elements !== prevProps.elements && this.props.namespace.name !== 'all') {
```

Comment:
Why are we excluding 'all' ?

Reply:
I'm relying on `this.props.namespace.name` to get health on services. I haven't found the `namespace` info available somewhere else. So obviously, 'all' didn't make it and resulted in empty health for all services.
I guess I should add `namespace` info in all graph nodes data from the backend. Unless someone sees another way to get the information.

But while thinking about it, maybe the other tasks about multi-namespaces feature already did that?

Reply:
I think is being worked here:

https://github.com/kiali/kiali/pull/252/files#diff-89df2dfb7238cfb78adc212f2a0581aeR35

Reply:
Oh, you're right, and as @jshaughn commented there we have the namespace as a part of "service" string. I'll fix that then.

Reply:
yes but that's not merged yet

Reply:
@josejulio it's something already there: namespace is included in the "fully qualified service name" that we get as input, such as 'productpage.bookinfo.blahblah' so let's use that.